### PR TITLE
For #23219 - Add favicons to synced tabs list

### LINKED ISSUE #23219: Favicons for synced tabs
Display favicons for synced tabs to make it easier to recognize sites.

### PR DESCRIPTION
**Pull request by MozillaNoah**
Closes #23219 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt [hunk 1]
```diff
@@ -210,31 +211,42 @@ fun SyncedTabsSectionHeader(
...
             .defaultMinSize(minHeight = 56.dp),
-        verticalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically,
     ) {
-        PrimaryText(
-            text = tabTitleText,
-            modifier = Modifier.fillMaxWidth(),
-            fontSize = 16.sp,
-            maxLines = 1
+        Favicon(
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:224]: The `@Preview` is broken from using `Favicon()`, so I'll be making a follow-up issue to refactor it to allow for previewing.

**Comment by gabrielluong** [comment]
@MozillaNoah Can you link the Figma?

**Comment by MozillaNoah** [comment]
> @MozillaNoah Can you link the Figma?

https://www.figma.com/file/hJfV1KGZuOK7OYr29pooL5/Synced-tabs?node-id=229%3A14717

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt [hunk 2]
```diff
@@ -210,31 +211,42 @@ fun SyncedTabsSectionHeader(
...
+    Row(
         modifier = Modifier
             .clickable(
                 onClickLabel = tabTitleText,
                 onClick = onClick
             )
             .padding(horizontal = 16.dp)
             .defaultMinSize(minHeight = 56.dp),
-        verticalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically,
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:222]: I see a 6dp top and 8dp bottom padding on inspection. What do you make of it versus using `CenterVertically`? Usually I go with the exact spec to align with what UX has designed.

Screen Shot 2022-03-09 at 6 17 02 PM

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt [hunk 3]
```diff
@@ -210,31 +211,42 @@ fun SyncedTabsSectionHeader(
...
+                modifier = Modifier.fillMaxWidth(),
+                fontSize = 16.sp,
+                maxLines = 1,
+            )
+
+            SecondaryText(
+                text = url,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 2.dp),
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:245]: I think this is okay, but makes me wonder if we should use a `Spacer`, if not when should we say to use a Spacer? An individual `Spacer` has been nice to explicitly call out spacing between 2 components.  Let me know what you decide.

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt [hunk 4]
```diff
@@ -210,31 +211,42 @@ fun SyncedTabsSectionHeader(
...
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(top = 2.dp),
-            fontSize = 12.sp,
-            maxLines = 1
-        )
+        Spacer(modifier = Modifier.width(32.dp))
+
+        Column(
+            verticalArrangement = Arrangement.Center,
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:232]: We shouldn't need this `verticalArrangement` if we add the top/bottom padding.

**Review by gabrielluong — APPROVED** [review]
r+ assuming we make the 6dp top and 8pd padding change. If not, ping me prior to landing.

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt (hunk 3 above)
Comment (MozillaNoah) [reply app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:245]: Yeah, I think this just didn't get the same treatment as other Composables in the file, so I'll go ahead and make the small refactor. I think it'll be helpful to distinguish true padding between elements (or within elements in the case of like some custom compound composable) and just whitespace (using a Spacer in this instance)

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt (hunk 2 above)
Comment (MozillaNoah) [reply app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:222]: the vertical alignment here is more so the icon is centered. If I use the padding values though, I may be able to remove the explicit `.defaultMinSize(minHeight = 56.dp)`

**Comment by gabrielluong** [comment]
To followup on everything, we got updated design specs from @cwzilla https://www.figma.com/file/ujdiReypSypu2GhRgzArea/branch/ddGnMfYggXGmcTYfoI0aS8/Mobile-Android-Components?node-id=1826%3A33615. Top and bottom padding are 10dp and we can remove the 2dp spacer in between the primary and secondary text.

**Pull request merged**